Update to E2E 1.xx

Discomanco opened the pull request:
First round of fixes.
So far, Cosmic Meatballs seems to potentially only happen on upgraded clients, needs further testing.
Dark Steel Upgrades is more complicated than anticipated
Unbreakable IE Rods is user error, not sure if fixable

Discomanco committed:
Merge branch 'EnigmaticaModpacks:master' into master

Discomanco committed:
Update .gitignore

Discomanco committed:
Update .gitignore

Discomanco committed:
Skullfire Sword fix

Removed enchanted books from Skullfire Sword recipe, as its impossible to future proof

Discomanco committed:
Revert "Update .gitignore"

This reverts commit 58a1f2b597ac3ded371368d79ed1a72cbb15790f.

Discomanco committed:
DE rock crystals

Fix to DE Rock Crystal Items having been uncraftable

Discomanco committed:
Merge branch 'develop' into master

Discomanco committed:
Remove Animania

Remove all traces of Animania

Discomanco committed:
Merge branch 'master' of https://github.com/Discomanco/Enigmatica2Expert

ReiDaTecnologia commented:
Rip Animania you will be slightly missed xD

Discomanco committed:
Add Anglesite and Benitoite

With ER getting removed, we need the gems back. Use ContentTweaker to do so

Discomanco committed:
Create Endergy.zs

Add Endergy, remove all lower tier items

Discomanco committed:
Removal of Extreme Reactors

Endergy is taking its place in many places, scripts has been adjusted for such.

Discomanco commented:
-Animania
-Extreme Reactors
+Endergy
Scripts from E2E-V2 used, still needs testing and polishing

ReiDaTecnologia commented:
Rip Extreme Reactors, you will not be missed xD

NielsPilgaard commented on .gitignore:
```diff
@@ -102,3 +102,7 @@ config/AppliedEnergistics2/VersionChecker.cfg
+config/vanillafix.cfg
+config/vanillafix/config_version
+config/vanillafix/methods-stable_39.csv
+config/splash.properties
```
we need that in I'm afraid, it configures the startup screen.

Discomanco replied:
Actually not sure when I added that to .gitignore, but apparently I did :P

NielsPilgaard commented on scripts/DraconicEvolution.zs:
```diff
@@ -11,7 +11,7 @@ import mods.jei.JEI.removeAndHide as rh;
 	<draconicevolution:grinder>, 
 	[[<ore:ingotElectricalSteel>, <draconicevolution:draconium_block:1>, <ore:ingotElectricalSteel>],
-	[<astralsorcery:itemcrystalsword>.anyDamage(), <draconicevolution:draconic_core>, <astralsorcery:itemcrystalsword>.anyDamage()], 
+	[<astralsorcery:itemcrystalsword>.onlyDamageAtLeast(0), <draconicevolution:draconic_core>, <astralsorcery:itemcrystalsword>.onlyDamageAtLeast(0)], 
```
Does this work? :D

Discomanco replied:
It's what is used for Rune of Sacrifice, and assuming the reason that one didn't break. I did test it out with freshly made swords and had no issue :D

NielsPilgaard commented on scripts/Endergy.zs:
```diff
@@ -0,0 +1,104 @@
+	
+	
+# Blutonium Block // Melodic Block
+//	recipes.remove(<minecraft:dirt>);
```
errr? :P

Discomanco replied:
Yeah, still need tidbits of code cleanup :P

NielsPilgaard commented on scripts/Endergy.zs:
```diff
@@ -0,0 +1,104 @@
+	[<ore:depletedFuelLEU235>, <actuallyadditions:block_crystal_empowered:1>, <ore:depletedFuelLEU235>]]);
+
+# Ludicrite Block // Stellar Block
+//	recipes.remove(<minecraft:blockmetals:4>);
```
I think you forgot to remove this.

Discomanco replied:
Polishing missing indeed :P

NielsPilgaard reviewed (approved):
Looks good to me, great work!

Discomanco committed:
Update .gitignore

Discomanco committed:
Update Endergy.zs

Clean up script

Discomanco committed:
Fixing broken Lapotron Crystals

Recipes only accepted fully charged Lapotron Crystals, this seems to fix it (same method used for Quantum Armor)

Discomanco committed:
Update Endergy.zs

Re-enable some items for Benitoite, making it require 15 items just like Anglesite

Discomanco commented:
Only thing left now is the Dark Steel Armor Upgrade recycling, but it has proven difficult to get around

NielsPilgaard commented:
> Only thing left now is the Dark Steel Armor Upgrade recycling, but it has proven difficult to get around

What seems to be going on there?

Discomanco commented:
> > Only thing left now is the Dark Steel Armor Upgrade recycling, but it has proven difficult to get around
> 
> What seems to be going on there?

I can't do a wildcard NBT `.anyDamage()`/`.withTag({})` with the machine (or Mek machines), and it also doesn't seem to want to accept an array of exact NBT tags of `val upgradesArray as IData = { "enderio:dsu":"enderio:<upgradeName>", ..` in a `for each/key/value/(key+value) in upgradesArray {`

NielsPilgaard commented:
> And I really don't feel like doing 27 different recipes for each individual recipe (double that if activated upgrades are to be included too)

Understandable! Would you prefer leaving them out or making the 27 recipes? 😄

Discomanco commented:
Going to double check with Mek machines, although hopes are low. And if not, I might use a tool for it. I kinda want it in :P

Discomanco committed:
Fix Ultimate Tank issue

Black Hole Tanks could only be crafted with an Ultimate Tank that had been placed in world, since it needed the `mekData:{}` tag.
This allows all 3 for the recipes

Discomanco committed:
Update Creative.zs

There was 2 buckets of Corium

Discomanco committed:
Fix DSU recycling

Fix and allow Dark Steel Upgrades to be recycled.
Some changes were made from test client, and is currently untested in the actual client

NielsPilgaard commented on scripts/ThermalExpansion.zs:
```diff
@@ -84,7 +84,7 @@ for slab in <ore:slabWood>.items {
 	<thermalexpansion:frame:132>, 
 	[<gendustry:genetics_processor>, <extrautils2:ingredients:5>, <ic2:plate:16>,
-	<ic2:lapotron_crystal>.anyDamage(), <forestry:thermionic_tubes:12>, <forestry:crafting_material:1>,
+	<ic2:lapotron_crystal:*>, <forestry:thermionic_tubes:12>, <forestry:crafting_material:1>,
```
DOES THIS WORK!? Could've sworn I tried that :P

Discomanco replied:
It's what's used for Quantum armor, and also why they didn't break :D

NielsPilgaard commented:
Great work Disco :D

Shijikori commented:
out of curiosity, because I returned to Enigmatica modpacks recently and noticed the removal of Animania... why was it removed? Performance/Stablity reasons?

NielsPilgaard commented:
Performance yeah 👍
